servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/Executors.java

```diff
@@ -48,6 +48,29 @@ public static Executor immediate() {
         return IMMEDIATE_EXECUTOR;
     }
 
+    /**
+     * Creates a new {@link Executor} that has a single dedicated worker thread. If {@link Executor#execute(Runnable)}
+     * method is invoked on the worker thread the {@link Runnable#run()} method will be executed immediately, but
+     * concurrent use is otherwise rejected.
+     *
+     * @return A new {@link Executor} that will use a single thread.
+     */
+    public static Executor newSingleThreadedExecutor() {
+        return from(new SingleThreadedExecutor());
+    }
+
+    /**
+     * Creates a new {@link Executor} that has a single dedicated worker thread. If {@link Executor#execute(Runnable)}
+     * method is invoked on the worker thread the {@link Runnable#run()} method will be executed immediately, but
+     * concurrent use is otherwise rejected.
+     *
+     * @param namePrefix A prefix for the worker thread name. A sequence number will always be appended.
+     * @return A new {@link Executor} that will use a single worker thread.
+     */
+    public static Executor newSingleThreadedExecutor(String namePrefix) {
```

Comment:
Are we confident that this is the approach to move forward? WDYT about exposing it only in some `internal` package or, let's say in this package but choose a different factory class: `ExperimentalExecutors`.
That way we can test how apps behave without risking that users depend on it.

Reply:
Certainly reasonable at this point. As long we can easily move or expose it later.